app/static/src/app/store/model/actions.ts

```diff
@@ -23,24 +66,20 @@ export const actions: ActionTree<ModelState, AppState> = {
     },
 
     async FetchOdin(context) {
-        const { rootState } = context;
-        const model = rootState.code.currentCode;
+        await fetchOdin(context);
+    },
 
-        await api(context)
-            .withSuccess(ModelMutation.SetOdin)
-            .withError(`errors/${ErrorsMutation.AddError}` as ErrorsMutation, true)
-            .post<OdinModelResponse>("/odin/model", { model });
+    CompileModel(context) {
+        compileModel(context);
     },
 
-    RunModel(context, payload: RunModelPayload) {
-        const { state, commit } = context;
-        if (state.odinRunner && state.odin) {
-            const {
-                parameters, start, end, control
-            } = payload;
+    RunModel(context) {
+        runModel(context);
+    },
 
-            const solution = state.odinRunner(dopri.Dopri, state.odin, parameters, start, end, control);
-            commit(ModelMutation.SetOdinSolution, solution);
-        }
+    async DefaultModel(context) {
+        await fetchOdin(context);
+        compileModel(context);
+        runModel(context);
     }
```

Comment:
```suggestion
    async DefaultModel(context) {
               await Promise.all([
            fetchOdin(context),
            compileModel(context),
            runModel(context)
        ]);
    }
```

Reply:
To avoid race issues

Reply:
No, we do need in these to happen in order - so we can't start Compile until after fetchOdin has completed (Promise.all fires all the promises off together) so we do need to await that separately. And as I say, I can't see any reason to make compile or run asynchronous.